### DISCUSSION TITLE
[Question]: How to set GOOGLE_REVERSE_PROXY？

### DISCUSSION THREAD

**Discussion by gordonchanfz** [danny-avila/LibreChat#1636] [Q&A] [answered]
### What is your question?

I want to use gemini pro。set the GOOGLE_REVERSE_PROXY in .env,but it doesnt work.

### More Details

Something went wrong. Here's the specific error message we encountered: [GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1/models/gemini-pro:streamGenerateContent?key=[REDACTED] [400 Bad Request] User location is not supported for the API use.

### What is the main subject of your question?

Installation

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
`GOOGLE_REVERSE_PROXY` is only supported for Vertex AI. The Google client for gemini has no mechanism to configure the baseURL

**Gemini uses the Gen Language API:**
https://docs.librechat.ai/install/configuration/ai_setup.html#generative-language-api-gemini

**Codey and PaLM use Vertex AI:**
https://docs.librechat.ai/install/configuration/ai_setup.html#vertex-ai-palm-2-codey

We need to update the docs to reflect this.

Soon you can access Gemini via Vertex AI, for which you can configure the reverse proxy url: https://github.com/langchain-ai/langchainjs/pull/3960

I'm also curious, what reverse proxy are you trying to set? For google specifically it must adhere to the Google API specs, not OpenAI

**Reply by gordonchanfz** [reply]
Tks for your reply.I will try after it is supported.
